Index: src/sys/netinet/in.c
===================================================================
RCS file: /cvs/src/sys/netinet/in.c,v
retrieving revision 1.162
diff -u -p -r1.162 in.c
--- src/sys/netinet/in.c	13 Feb 2019 23:47:43 -0000	1.162
+++ src/sys/netinet/in.c	23 Feb 2019 19:08:58 -0000
@@ -103,7 +103,7 @@ in_canforward(struct in_addr in)
 {
 	u_int32_t net;
 
-	if (IN_EXPERIMENTAL(in.s_addr) || IN_MULTICAST(in.s_addr))
+	if (IN_MULTICAST(in.s_addr))
 		return (0);
 	if (IN_CLASSA(in.s_addr)) {
 		net = in.s_addr & IN_CLASSA_NET;
