From 385793cad5af9fd762bb48e4d8571101c6ff7c93 Mon Sep 17 00:00:00 2001
From: Seth Schoen <schoen@loyalty.org>
Date: Mon, 7 Dec 2020 16:09:17 -0800
Subject: [PATCH] Reduce special cases for IPv4 address space

---
 libbb/atlas_check_addr.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/libbb/atlas_check_addr.c b/libbb/atlas_check_addr.c
index 61dfa4b..d7f063c 100644
--- a/libbb/atlas_check_addr.c
+++ b/libbb/atlas_check_addr.c
@@ -12,13 +12,11 @@ struct ipv4_prefix
 }
 static bad_ipv4[] =
 {
-	{ 0x00000000,  8 },	/*     0.0.0.0/8 null */
-	{ 0x7F000000,  8 },	/*   127.0.0.0/8 localhost */
+	{ 0x7F000000, 16 },	/*   127.0.0.0/16 localhost */
 	{ 0x0A000000,  8 },	/*    10.0.0.0/8 (RFC-1918) */
 	{ 0xAC100000, 12 },	/*  172.16.0.0/12 (RFC-1918) */
 	{ 0xC0A80000, 16 },	/* 192.168.0.0/16 (RFC-1918) */
 	{ 0xA9FE0000, 16 },	/* 169.254.0.0/16 (RFC-3927) */
-	{ 0xE0000000,  3 },	/*   224.0.0.0/3 multicast and reserved */
 };
 
 struct ipv6_prefix
-- 
2.17.1

